job_templates/capsule_upgrade_-_ansible_default.erb

```diff
@@ -0,0 +1,42 @@
+<%#
+name: Smart Proxy Upgrade Playbook
+snippet: false
+template_inputs:
+- name: target_version
+  required: true
+  input_type: user
+  advanced: false
+  value_type: plain
+  hidden_value: false
+- name: whitelist_options
+  required: false
+  input_type: user
+  advanced: false
+  value_type: plain
+  hidden_value: false
+model: JobTemplate
+job_category: Ansible Playbook
+description_format: "%{template_name}"
+provider_type: Ansible
+kind: job_template
+%>
+
+---
+- hosts: all
+  tasks:
+    - block:
+      - name: Upgrade Smart Proxy using foreman-maintain
+        command: foreman-maintain upgrade run --assumeyes --target-version=<%= input('target_version') %> --whitelist="<%= input('whitelist_options') %>"
```

Comment:
Does foreman-maintain detect that it is upgrading a Capsule and not Satellite?

Reply:
yes it automatically identifies Satellite vs. Capsule, its the same command for both

Reply:
Does it identify it correctly if I run it against the internal capsule?

Reply:
you mean if this playbook differentiates if it is Satellite or Capsule server? It doesnt. If needed, it is trivial to add - just check for the presence of `satellite` or `satellite-capsule` rpm.

Reply:
now added a check to see if the target server is `satellite` and abort action.